Examples/Multiplayer/CSharp/Foundation-Unity-Example/Assets/Scripts/HelloWorld.cs

```diff
@@ -0,0 +1,649 @@
+        if (!successShutdown)
+        {
+            Debug.Log("Failed to shut down Foundation. Error is within Foundation package.");
+            return;
+        }
+
+        Debug.Log("Shutdown Csp Foundation");
+        foundationHasStarted = false;
+    }
+
+    private void QuitFoundation()
+    {
+        Application.quitting -= QuitFoundation;
+
+        if (foundationHasStarted)
+        {
+            if (enteredSpace)
+            {
+                ExitSpaceAsync();
+            }
+
+            StopFoundation();
+        }
+    }
+
+    #endregion
+
+    // Authentication / Login
+
+    #region Authentication & Flow
+
+    /// <summary>
+    /// Listener to trigger creating a new account based on UI input for the user using the given email and password.
+    /// </summary>
+    /// <param name="email"> Email of the new account. </param>
+    /// <param name="password"> Password of the new account. </param>
+    private async void SignUpAsync(string email, string password)
+    {
+        await CreateUserAsync(email, password);
+    }
+
+    /// <summary>
+    /// Listener to trigger signing in with an account based on UI input for the user using the given email and password.
+    /// </summary>
+    /// <param name="email"> Email of the new account. </param>
+    /// <param name="password"> Password of the new account. </param>
+    private async void SignInAsync(string email, string password)
+    {
+        await LoginAsync(email, password);
+        await SearchSpacesAsync();
+        await SearchSpacesUsingGraphQLAsync();
+    }
+
+    /// <summary>
+    /// Requests the Foundation layer to create a new account for the user using the given email and password.
+    /// </summary>
+    /// <param name="email"> email of the new account.</param>
+    /// <param name="password"> password of the new account.</param>
+    /// <returns> Just the Task object to await.</returns>
+    private async Task CreateUserAsync(string email, string password)
+    {
+        Debug.Log($"Creating account with Email {email} ...");
+
+        using CspSystems.ProfileResult result =
+            await userSystem.CreateUser(null, null, email, password, false, true, null, null);
+        CspSystems.Profile profile = result.GetProfile();
+
+        Debug.Log($"Created account with Email {profile.Email}. Please check your email to verify your account.");
+    }
+
+    /// <summary>
+    /// Requests the Foundation layer to Login with email and password.
+    /// </summary>
+    /// <param name="email"> Email of the user. </param>
+    /// <param name="password"> Password of the user. </param>
+    /// <returns> Just the Task object to await.</returns>
+    private async Task LoginAsync(string email, string password)
+    {
+        Debug.Log($"Logging in with Email {email} ...");
+
+        using CspSystems.LoginStateResult loginResult = await userSystem.Login(string.Empty, email, password, true);
+
+        // Cache user ID for later use.
```

Comment:
Would it be possible to also add this fix to the basic framework example ? as it would be an issue there. As far as i know the only place it's needed there is for the avatar creation.

Reply:
sure. I can add this fix to the basic framework example.